content/en/platform/corda/5.1/release-notes/5.1.0.md

```diff
@@ -11,6 +11,164 @@ section_menu: corda51
 ---
 # Corda 5.1.0 Release Notes
 
+This page lists the key changes in the 5.1 release of Corda and contains the following:
 
+* [Enhancements](#enhancements)
+* [Fixed Issues](#fixed-issues)
+* [Known Issues and Limitations](#known-issues-and-limitations)
+
+For information about upgrading a Corda Cluster from 5.0 to 5.1, see [Upgrading from 5.0]({{< relref "../deploying-operating/deployment/upgrading/_index.md" >}}).
+
+## Enhancements
+
+### Architecture
+
+#### Java 17
+
+Corda components are now hosted in a Java 17 compatible JVM.
+
+#### Workers
+
+The following new types of workers have been added:
+
+* Persistence workers
+* Uniqueness workers
+* Flow mapper workers
+* Verification workers
+* Token selection workers
+
+#### Logging
+
+The default log4j configuration was modified to print exception stack traces as text blob within the JSON log message.
+
+### CorDapp Development
+
+#### Logging for Finality Flows
+
+The ID in logs written for finality flows has been made consistent.
+
+#### Group Parameters
+
+A new interface, `GroupParametersLookup`, has been added to the `net.corda.v5.membership` package to enable flows to access `GroupParameters` for further verification.
+
+#### findUnconsumedStatesByType Performance
+
+The performance of the retrieval of unconsumed states in the `findUnconsumedStatesByType` query has been improved. As a result, it it is now more efficient to scan the table with an index on `utxo_visible_transaction_state.consumed`.
```

Comment:
This function gets deprecated with 5.1. Probably, I would not mention it here.

Reply:
Maybe we shold rather mentioned that it has been deprecated for removal in 5.2? I am not even sure its performance has been improved?

Reply:
This was added for this ticket: https://r3-cev.atlassian.net/browse/CORE-15504

Reply:
Sorry, that was a misunderstanding presented by JIRA... one of those is a sub-task of the other, and they really are the same story.
Yes, `findUnconsumedStatesByExactType` replaces `findUnconsumedStatesByType`, we couldn't improve the performance without changing the behaviour.

Reply:
Updated 👍